### PR TITLE
share process hook and image calling api

### PR DESCRIPTION
**Pull request by arealclimber**
### DEVELOP

- [x]  /share/cfd/[cfdId].tsx
- [x] /api/images/cfd/[cfdId].tsx
- [x] HistoryPositionModal
- [x] globalContext
- [x] useShareProcess
- [x] TradeBook

### CHECK LIST
- [x]  #676 
- [x] #644 
- [x] #690 
- [x] #691 
- [x] 改回 DOMAIN
- [x] Demo 完後將 enableShare 改成正常狀態
- [Naming convention](https://github.com/CAFECA-IO/WorkGuidelines/blob/main/technology/coding-convention/naming-convention.md) verification: checked
- Coding style verification: checked
- new Library: 0
- new Class / Component:  0
- new loop: 0
- new recursive: 2
    - src/lib/hooks/use_share_process.ts (132, 159)
- high risk: 0
- new sql: 0

### UML

- [Sequence Diagram 001 (interaction between user, frontend and backend)](https://github.com/CAFECA-IO/Documents/blob/main/TBD/TBDSD00001.md)
- [Sequence Diagram 002 (background mechanism)](https://github.com/CAFECA-IO/Documents/blob/main/TBD/TBDSD00002.md)
- [Component Diagram](https://github.com/CAFECA-IO/Documents/blob/main/TBD/TBDCP00001.md)
- [Class Diagram](https://github.com/CAFECA-IO/Documents/blob/main/TBD/TBDCD00001.md)
- [Global context document](https://www.notion.so/Global-Context-public-d93ac7c941c94fb6b173d34a7bc4e959)
- [Table of interface](https://github.com/CAFECA-IO/EventReport/blob/main/meeting/20230106_tbd_mvp.md)

### CODE REVIEW

File: src/pages/api/images/cfd/[cfdId].tsx [hunk 1]
```diff
@@ -39,31 +42,68 @@ export default async function handler(req: NextApiRequest, res: NextApiResponse)
...
-    closeTime,
-  } = getDummySharingOrder() as ISharingOrder;
+    createTimestamp,
+    closeTimestamp,
+  } = sharingOrder;
 
-  const {date: openDate, time: openTimeString} = timestampToString(openTime);
-  const {date: closeDate, time: closeTimeString} = timestampToString(closeTime);
+  const {date: openDate, time: openTimeString} = timestampToString(createTimestamp);
+  const {date: closeDate, time: closeTimeString} = timestampToString(closeTimestamp);
```
Comment (arealclimber) [inline src/pages/api/images/cfd/[cfdId].tsx:96]: - 標明現在拿到的時區是 UTC+n

**Review by arealclimber — COMMENTED** [review]
code review

File: src/pages/share/cfd/[cfdId].tsx [hunk 2]
```diff
@@ -35,11 +35,16 @@ const CfdSharing = (props: IPageProps) => {
   const appCtx = useContext(AppContext);
   const router = useRouter();
 
-  const {cfdId} = router.query;
+  const [userTz, setUserTz, userTzRef] = useStateRef<number>(0);
+
+  // Deprecated: called by component by `fetch` won't have any query or props unless type something on the url of the browser (20230523 - Shirley)
```
Comment (arealclimber) [inline src/pages/share/cfd/[cfdId].tsx:40]: 刪掉 Deprecated

File: src/lib/common.ts [hunk 3]
```diff
@@ -44,12 +44,18 @@ export function getDeadline(deadline: number) {
   return Math.ceil(new Date().getTime() / 1000) + deadline;
 }
 
+export const adjustTimestamp = (timestamp: number, timezoneOffset: number) => {
```
Comment (arealclimber) [inline src/lib/common.ts:47]: get server timestamp

File: src/lib/hooks/use_share_process.ts [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+/**
+ * @param lockerName: unique name, string `'filename_functionname'` `e.g.'history_position_modal.shareHandler'`
+ */
+interface IUseShareProcess {
+  lockerName: string;
+  shareType: IShareType;
+  shareId: string;
+  enableShare: (id: string, share: boolean) => Promise<IResult>;
+}
+
```
Comment (arealclimber) [inline src/lib/hooks/use_share_process.ts:29]: 備註為了檢查 order 是否一致
``` 

cfd: ICfd

```

File: src/pages/api/market/trades/index.ts [hunk 5]
```diff
@@ -29,4 +29,5 @@ export default async function handler(req: NextApiRequest, res: NextApiResponse)
       res.status(500).json({error: Reason[Code.INTERNAL_SERVER_ERROR]});
     }
   } else res.status(500).json({error: Reason[Code.INTERNAL_SERVER_ERROR]});
+  return;
```
Comment (arealclimber) [inline src/pages/api/market/trades/index.ts:32]: - rm

**Review by Luphia — APPROVED** [review]
- no bug found

**Pull request merged**